exercises/binary-search/src/test/java/BinarySearchTest.java

```diff
@@ -0,0 +1,100 @@
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class BinarySearchTest {
+
+    private static final List<Integer> SORTED_LIST
+            = Collections.unmodifiableList(
+                    Arrays.asList(1, 2, 3, 4, 5, 6)
+            );
+
+    public static final List<Integer> SORTED_LIST_OF_ODD_LENGTH
+            = Collections.unmodifiableList(
+                    Arrays.asList(0, 1, 2, 2, 3, 10, 12)
+            );
+
+    public static final List<Integer> UNSORTED_LIST
+            = Collections.unmodifiableList(
+                    Arrays.asList(10, 2, 5, 1)
+            );
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
```

Comment:
Nice use of Rules! Since this is a relatively advanced concept that we've only just begun integrating into the track, we've been marking all usages with a helpful comment (for an example, see https://github.com/exercism/xjava/blob/master/exercises/queen-attack/src/test/java/QueenAttackCalculatorTest.java#L11-L14). Please add that same comment here. Once we have established a progression of exercises per #142, then we'll likely reduce this redundancy some, but better safe than sorry for now.